portal/src/axios.js

```diff
@@ -1,190 +1,23 @@
-/**
- * This file is the axios.js file that gets generated by NuxtJS.
- * We should see which configurations we want to keep here and remove the rest.
- */
-
 import Axios from 'axios'
-import { isNil } from 'lodash'
-
-// Axios.prototype cannot be modified
-const axiosExtra = {
-  setHeader(name, value, scopes = 'common') {
-    for (let scope of Array.isArray(scopes) ? scopes : [scopes]) {
-      if (!value) {
-        delete this.defaults.headers[scope][name]
-        return
-      }
-      this.defaults.headers[scope][name] = value
-    }
-  },
-  setToken(token, type, scopes = 'common') {
-    const value = !token ? null : (type ? type + ' ' : '') + token
-    this.setHeader('Authorization', value, scopes)
-  },
-  setLanguage(language) {
-    if (!isNil(language)) {
-      this.defaults.headers.common['Accept-Language'] = language
-    } else {
-      delete this.defaults.headers.common['Accept-Language']
-    }
-  },
-  onRequest(fn) {
-    this.interceptors.request.use(config => fn(config) || config)
-  },
-  onResponse(fn) {
-    this.interceptors.response.use(response => fn(response) || response)
-  },
-  onRequestError(fn) {
-    this.interceptors.request.use(
-      undefined,
-      error => fn(error) || Promise.reject(error)
-    )
-  },
-  onResponseError(fn) {
-    this.interceptors.response.use(
-      undefined,
-      error => fn(error) || Promise.reject(error)
-    )
-  },
-  onError(fn) {
-    this.onRequestError(fn)
-    this.onResponseError(fn)
-  }
-}
-
-// Request helpers ($get, $post, ...)
-for (let method of [
-  'request',
-  'delete',
-  'get',
-  'head',
-  'options',
-  'post',
-  'put',
-  'patch'
-]) {
-  axiosExtra['$' + method] = function() {
-    return this[method].apply(this, arguments).then(res => res && res.data)
-  }
-}
-
-const extendAxiosInstance = axios => {
-  for (let key in axiosExtra) {
-    axios[key] = axiosExtra[key].bind(axios)
-  }
-}
+import i18n from './i18n'
 
-const setupProgress = axios => {
-  if (process.server) {
-    return
-  }
+const baseURL = process.env.VUE_APP_BACKEND_URL + 'api/v1'
+const axios = Axios.create({ baseURL })
+axios.defaults.xsrfCookieName = 'csrftoken'
+axios.defaults.xsrfHeaderName = 'x-csrftoken'
 
-  // A noop loading inteterface for when $nuxt is not yet ready
-  const noopLoading = {
-    finish: () => {},
-    start: () => {},
-    fail: () => {},
-    set: () => {}
+axios.interceptors.response.use(function(response) {
+  if (response && response.data) {
+    return response.data
```

Comment:
I don't agree with this. We should have response status codes as well as headers. We can't possibly say now that we'll never need them. If stores want data they should just get it themselves instead of feeding everything only data as if nothing else ever matters.

Reply:
I don't agree with it either. However the whole application was built like this (however it was hidden in some obscure monkey-patched methods which were used everywhere). We might want to change this behaviour, but that's a lot more of work.

You can always eject an interceptor when you need something like status code or headers. Now we don't use it anywhere, so I prefer to keep it as simple as possible.

Reply:
We should keep the $ methods for now and then migrate away step by step I think. Whenever you touch a file using axios.$ you change it to the normal method and rewrite to use .data. It adds 7 lines of code to axios file for now. When I talk about confusing code and keeping things simple I mean for instance the filters-categories component. Having to eject based on context sounds like more obscure code all over the place instead of adding a bit of functionality in only one file for now.